lib/handler/headers_util.c

```diff
@@ -4,26 +4,120 @@ static h2o_header_t *find_header(h2o_headers_t *headers, h2o_headers_command_t *
 {
     ssize_t index;
 
-    if (h2o_iovec_is_token(cmd->name)) {
-        index = h2o_find_header(headers, (void *)cmd->name, -1);
+    assert(cmd->num_args == 1);
+    if (h2o_iovec_is_token(cmd->args[0].name)) {
+        index = h2o_find_header(headers, (void *)cmd->args[0].name, -1);
     } else {
-        index = h2o_find_header_by_str(headers, cmd->name->base, cmd->name->len, -1);
+        index = h2o_find_header_by_str(headers, cmd->args[0].name->base, cmd->args[0].name->len, -1);
```

Comment:
I like the idea of making sure that `find_header` can be used for handling commands that take only one name as the argument.

Would it make sense to change the signature of the function to `h2o_header_t *find_header(h2o_headers_t *headers, h2o_iovec_t *name)` so that the API implies how it should be used?